### ISSUE TITLE
Retrospective

### ISSUE THREAD

**Issue by DavidL0914** [DavidL0914/TheIdealGam#5]
Reflecting on the project

**Comment by DavidL0914** [comment]
My comments / grade: https://github.com/DavidL0914/TheIdealGam/issues/3#issuecomment-1850683525

**Retrospect:**
One of the main points my peer grader brought up was that I didn't show progress through issues or comments on my code throughout the project. I often committed lots of things at once instead of committing things after every small feature I add, making it look like I copied code (even though I didn't). So in the future, I should commit much more frequently, even if the things I add are small, and show my progress through issues or utterances. 

I think the CSS behind my page looks pretty good and matches the overall theme of our site. I think the "game feature" behind my feature is also really cool and helps teach people binary in a competitive way. However, I think I could have made the game more interesting by adding different types of binary questions instead of just converting decimal numbers to binary numbers. I also should add tips or instructions on how to convert decimals to binary numbers for players who are just starting to learn conversions. 

I also need to edit the way I make my demo CPT video. I need to not do any mouse wiggling and focus instead of editing to make clear to the viewer what my features are and what they do. 

**Conclusion:** 
Overall, I need to commit more frequently and update and show my progress through issues and utterances, commenting on things I changed from my initial plan. I also need to make the game more interesting with new features and think through the minds of my users, especially those new to the topic of binaries. Also, make demo video clearer and NO MOUSE WIGGLING!

**Comment by ak146** [comment]
[Grade](https://github.com/DavidL0914/TheIdealGam/issues/3#issuecomment-1850706055)
Answering complaints: The main problem that Trevor expressed was the difficulty of the projects without being taught the actual Ascii values for each letter. In order to counteract this I added a map of every letter in the alphabet. I also limited the ASCII guesser to only use letters from the alphabet to make it easier to follow. I also reorganized the buttons so that it shows the full process of converting words into binary. He also mentioned that the challenge button made less sense so I decided to have it combine the steps that were shown by the other two buttons into one step. I also made it so that the game does not keep you in an infinite loop if you are unable to guess the value. This was a complaint made by both Trevor and Mr. Mortensen.

For next time: I think the actual game feature was not very creative. It was difficult for me to think of any entertaining games that made use of binary as the main function. There were better ways to express the changes through the buttons and the version I presented was highly simplified. I should have made my code help memorize values instead of just having a person guess blindly. There should have also been more CSS in involved. My page looked very boring and could have been helped by making the features more interactive on the page.

I need to commit more frequently and use issues to explain my progress. I should also make my commit names more detailed instead of "aha." The feature should be more creative but I think the actual code worked well.

**Comment by 7mwang** [comment]
[Comment/Grade](https://github.com/DavidL0914/TheIdealGam/issues/3#issuecomment-1850702395)
**Retrospect:**
Looking back, there were a couple things that I didn't end up finishing that would have increased the quality of our project. Firstly, mentioned by Trevor in our team review, we didn't really have color codes implemented into our site. I intended to implement something like this into my feature, but didn't end up doing so. I should have dedicated more time to thinking about how to incorporate more features of binary/hexadecimal.

In my video, Mr. Mortensen also called out that a lot of peoples' captions were too fast. This 100% applied to me, and it was not my intention to have such short read-times. Regardless, it was my mistake, and I should have spent less time on less important features, and spent more time with longer-lasting and more readable comments on the more important features. Other than my fast-paced comments, I think I did alright demonstrating my feature, at least having arrows pointing to what was popping up instead of mouse wiggling, and having captions on the bottom to show overall what was going on.

Lastly, I know I could have fleshed out my code more. It had a basic function, but I think I could have had more, touching back on the comment about not implementing more features. For example, I could have had any form of color customization on the site that used a binary to hexadecimal conversion system.

**Summary:**
In summary, I could have added more to my feature in the form of logic gates/hexadecimal, and I could have improved the quality of my video by making the comments longer-lasting. I think the basic function of my project was well done, but I could have added more depth to it.

**Comment by KaydenLe** [comment]
My comments / Grade: [#3 comment](https://github.com/DavidL0914/TheIdealGam/issues/4#issuecomment-1852610661)

Answering complaints: The main problem that Hanlun said that it was difficult to see the letters and the instructions for the game because the words were in a dark font he suggested that the words were to be in a brighter color. He also suggested that there could instructions and a warning of guessing some numbers is harder than others. He also suggested a leader board and timer. Overall his complaints were suggested but his biggest complaint was changing the font color of the instructions so it is visually easy to see. 

For the future: The feature was a very basic feature. It wasn't entertaining. Even though it works very well it could be better and there are other ways to use binary in a feature and also have the instructions and words be easy to see by using a brighter color.

Summary: I could have added more meaningful commit names instead of naming them all changes and have a more creative idea behind the feature instead of something basic.
